install/kubernetes/cilium/templates/hubble-ui/deployment.yaml

```
@@ -102,7 +102,7 @@ spec:
         - name: TLS_TO_RELAY_ENABLED
           value: "true"
         - name: TLS_RELAY_SERVER_NAME
-          value: ui.hubble-relay.cilium.io
+          value: {{ .Values.hubble.relay.tls.server.clusterDomain }}
```

Almost. We should only use the clusterDomain for the cluster domain part. The rest should be derived differently:

```suggestion
          value: relay.{{ .Release.Namespace }}.svc.{{ Values.hubble.relay.tls.server.clusterDomain }}
```

I'm not sure if the default TLS name (`ui.hubble-relay.cilium.io`) is even correct. I'll ping some Hubble folks regarding this.

---

Okay, so it seems our Relay server certificates are currently hard-coded to use `*.hubble-relay.cilium.io`:
https://github.com/cilium/cilium/blob/44698dfecdd65788c621c705581cbf92ec46e3fd/install/kubernetes/cilium/templates/hubble/tls-helm/relay-server-secret.yaml#L3

That makes it impossible to use the `clusterDomain`, because we will want to have backwards-compatibility with existing certs. On the other hand, I'm also not a  fan of introducing a value for GKE datapath v2 that cannot be used with Cilium OSS.

For now, I'm fine accepting a patch which basically is the same as your original PR: A simple Helm value that only works on GKE datpath v2. It should be clear that this is a workaround and might be removed in future versions once we have a better solution.

Long-term however, I think a proper solution would be to fix up our all our certificate templates (`tls-cronjob`, `tls-helm`, `tls-certmanager`) to assemble the CommonName of the certificates from a `clusterDomain` and namespace, with some option to override it too.

---

@gandro - I have reverted the changes back to a simple key/value that van be customised and run the relevant make commands to generate the docs.  Let em know if there is anything else that should be changed.

---

We could pass this to `tpl` so you can use template vars.

eg:
```
clusterDomain: "relay.{{ .Release.Namespace }}.svc.{{ Values.hubble.relay.tls.server.clusterDomain }}"
```

The template:
```
        - name: TLS_RELAY_SERVER_NAME
          value: {{ .Values.hubble.relay.tls.server.clusterDomain | tpl }}
```